presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java

```diff
@@ -1169,4 +1178,152 @@ public void testInsertForDefaultColumn()
     }
 
     protected abstract TestTable createTableWithDefaultColumns();
+
+    @Test(dataProvider = "testDataMappingSmokeTestDataProvider")
+    public void testDataMappingSmokeTest(DataMappingTestSetup dataMappingTestSetup)
+    {
+        String prestoTypeName = dataMappingTestSetup.getPrestoTypeName();
+        String sampleValueLiteral = dataMappingTestSetup.getSampleValueLiteral();
+        String highValueLiteral = dataMappingTestSetup.getHighValueLiteral();
+
+        String tableName = "test_data_mapping_smoke_test_" + prestoTypeName.replaceAll("[^a-zA-Z0-9]", "_");
+
+        Runnable setup = () -> {
+            String createTable = format("CREATE TABLE %s(id bigint, value %s)", tableName, prestoTypeName);
+            assertUpdate(createTable);
+            assertUpdate(
+                    format("INSERT INTO %s VALUES (10000, NULL), (10001, %s), (99999, %s)", tableName, sampleValueLiteral, highValueLiteral),
+                    3);
+        };
+        if (dataMappingTestSetup.isUnsupportedType()) {
+            String typeNameBase = prestoTypeName.replaceFirst("\\(.*", "");
+            String expectedMessagePart = format("(%1$s.*not (yet )?supported)|((?i)unsupported.*%1$s)|((?i)not supported.*%1$s)", Pattern.quote(typeNameBase));
+            assertThatThrownBy(setup::run)
```

Comment:
s/setup::run/setup?

Reply:
`ThrowingCallable` is not assignable from `Runnable`